### PR TITLE
Issue #16: started configuring logback logging

### PR DESCRIPTION
**Pull request by artemlisikh**
but didn't figure out how to remove deprecated logging info yet

### CODE REVIEW

File: src/main/resources/logback-spring.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (olisikh) [inline src/main/resources/logback-spring.xml]: I propose to have logback.groovy instead of XML, XML is shit!

Comment (olisikh) [reply src/main/resources/logback-spring.xml]: Nice that you're using spring boot integration for logging, let's stick to it. Here's a page, we need to add groovy dependency and migrate logback-spring.xml to logback.groovy

Here's a link: https://docs.spring.io/spring-boot/docs/current/reference/html/howto-logging.html#howto-configure-logback-for-logging

In groovy file, you'll be able to access System.getProperty("LOG_FILE") which would get you this system property, we don't want to use maven filtering here (@ logging.file @) as it's quite dangerous, we'll only use it for application.yml.

File: pom.xml [hunk 2]
```diff
@@ -23,12 +23,13 @@
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <java.version>1.8</java.version>
         <flyway.version>4.2.0</flyway.version>
+        <logging.file>logs/spring-boot-logging.log</logging.file>
```
Comment (olisikh) [inline pom.xml]: let's name it ${project.name}.log, what do you think?

File: src/main/resources/logback-spring.xml (hunk 1 above)
Comment (artemlisikh) [reply src/main/resources/logback-spring.xml]: Great idea! I totally support that.

File: pom.xml (hunk 2 above)
Comment (artemlisikh) [reply pom.xml]: My thoughts exactly

File: src/main/resources/logback-spring.xml (hunk 1 above)
Comment (olisikh) [reply src/main/resources/logback-spring.xml]: You're my star!

**Review by artemlisikh — COMMENTED** [review]
replaced ... with *

**Comment by artemlisikh** [comment]
what has just happened

File: pom.xml [hunk 3]
```diff
@@ -29,6 +29,9 @@
         <db.username>root</db.username>
         <db.password>5472</db.password>
         <spring.profiles.active>default</spring.profiles.active>
+
+        <logging.file>${project.name}.log</logging.file>
```
Comment (olisikh) [inline pom.xml]: I think we need to have logging.file only as a ${project.name}. Why? See next comments for logbackx.groovy

File: src/main/resources/logbackx.groovy [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    encoder(PatternLayoutEncoder) {
+        pattern = "%d{HH:mm:ss} %-4relative [%thread] %-5level %logger{35} - %msg%n"
+    }
+}
+
+appender("FILE", RollingFileAppender) {
+    file = "${LOG_PATH}/${LOG_FILE}"
+
+    rollingPolicy(TimeBasedRollingPolicy) {
+        fileNamePattern = "${LOG_ARCHIVE}/rollingfile.log%d{yyyy-MM-dd}.log"
```
Comment (olisikh) [inline src/main/resources/logbackx.groovy]: I think we need to have a name of the file as "${LOG_ARCHIVE}/${LOG_FILE}-%d{yyyy-MM-dd}.log" and this is why I think we should have LOG_FILE without extension.

File: src/main/resources/logbackx.groovy [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+
+appender("CONSOLE", ConsoleAppender) {
+    encoder(PatternLayoutEncoder) {
+        pattern = "%d{HH:mm:ss} %-4relative [%thread] %-5level %logger{35} - %msg%n"
+    }
+}
+
+appender("FILE", RollingFileAppender) {
+    file = "${LOG_PATH}/${LOG_FILE}"
```
Comment (olisikh) [inline src/main/resources/logbackx.groovy]: If we change LOG_FILE to be extensionless, we need to add .log to the end of this string.

File: src/main/resources/logbackx.groovy (hunk 4 above)
Comment (konteno) [reply src/main/resources/logbackx.groovy]: Can't apply that pattern because there's no filenamepattern for log file, only for archived log files

**Review by olisikh — APPROVED** [review]
Brilliant! Great job!

**Pull request merged**